src/modules/hg_branch.rs

```diff
@@ -74,13 +75,34 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     Some(module)
 }
 
-fn get_hg_branch_name(ctx: &Context) -> String {
-    std::fs::read_to_string(ctx.current_dir.join(".hg").join("branch"))
+fn find_dothg_path(cwd: &PathBuf) -> Option<PathBuf> {
```

Comment:
Does mercurial support something like `$GIT_CEILING_DIRECTORIES` (#3591)? It would be nice to avoid crossing devices at least.

Reply:
Unfortunately, it does not. This is the mercurial python code that locates the `.hg` directory.

```
def findrepo(p):
    while not os.path.isdir(os.path.join(p, b".hg")):
        oldp, p = p, os.path.dirname(p)
        if p == oldp:
            return None
    return p
```

Reply:
In that case, please change the implementation to stay on the same device if possible, and perhaps split that off into a new PR. 

It also looks like your hg topic changes broke some of the existing tests. Please fix that.